detox/src/devices/drivers/android/genycloud/services/dto/GenyInstance.js

```diff
@@ -0,0 +1,35 @@
+const _ = require('lodash');
+const Recipe = require('./GenyRecipe');
+
+class GenyInstance {
+  constructor(rawInstance) {
+    Object.assign(this, _.pick(rawInstance, 'uuid', 'name', 'state'));
+    this.adb = {
+      name: rawInstance.adb_serial,
+      port: rawInstance.adb_serial_port,
+    }
+    this.recipe = new Recipe(rawInstance.recipe);
+  }
+
+  isAdbConnected() {
+    return this.adb.name !== '0.0.0.0';
+  }
+
+  isTerminated() {
+    return this.state === 'RECYCLED';
```

Comment:
It won't hurt for the next generations of humans to know, which states it can have, at least from what you have found in your investigations. Make sure to put some JSDoc for `state` property there.

Reply:
See #2577